src/elements/fields/TextField.jsx

```diff
@@ -87,8 +125,145 @@ function TextField({
     inputRef,
     ...fieldProps
 }) {
+    // const anyCharRegex = '^\\w$';
     const servar = element.servar;
+    const {
+        fieldMaskRegex,
+        fieldMaskComparisionString,
+        fieldMaskString,
+        deterministicPattern,
+        maxRawInputLength
+    } = getFieldMaskMeta('$ {{ \\d{4} }} / year');
+
+    const [rawFieldValue, setRawFieldValue] = useState('');
+    const patternFieldMask = useRef(fieldMaskComparisionString);
+    const fieldValueMask = useRef(fieldMaskString);
+    const rawCaretPos = useRef(0);
+    const maskedCaretPos = useRef(4);
+
     const inputType = fieldProps.as === 'textarea' ? 'textarea' : 'input';
+
+    useHotkeys(
```

Comment:
could you add comments for what each of these keyboard listeners do?

Reply:
done.